### PR TITLE
fix: wakunode2 config. adding new 'topic' config parameter.

### LINKED ISSUE #1726: bug: inconsistent usage of topic flags
### Problem
There's two ways to use topics flags:

* `--topics` - Takes a list of comma separated topics, CANNOT be repeated.
* `--protected-topic` - Does NOT take a list, CAN be repeate.

Makes no sense.

### Impact
Almost made me vomit.

### Expected behavior
Not inducing nauseousness. Probably just make `--topics` into `--topic` and allow it to be repeated.

### Additional context
https://github.com/waku-org/nwaku/blob/b277ce1013563b4e76a3f8694f0eefb1f3528a4d/apps/wakunode2/config.nim#L210-L213
https://github.com/waku-org/nwaku/blob/b277ce1013563b4e76a3f8694f0eefb1f3528a4d/apps/wakunode2/config.nim#L40-L43

### PR DESCRIPTION
**Pull request by Ivansete-status**
# Description
Adding a new `topic` parameter can be repeated.
The `topics` parameter becomes deprecated because it is not consistent with the other "repeatable" ones as it expects a space-separated list of pubsub topics.

## How to use the new parameter
When setting different pubsub topics through the CLI, they can be repeated:
`./build/wakunode2 --topic=t1 --topic=t2 --topic=t3 ...`

However, when setting them from within a config file, they should be in an array format within the cfg file:
```
topic = [ "t1" "t2" "t3" ]
```

## Issue
closes #1726

### CODE REVIEW

File: apps/wakunode2/app.nim [hunk 1]
```diff
@@ -547,7 +547,16 @@ proc setupProtocols(node: WakuNode, conf: WakuNodeConf,
     peerExchangeHandler = some(handlePeerExchange)
 
   if conf.relay:
-    let pubsubTopics = conf.topics.split(" ")
+
+    var pubsubTopics = @[""]
+    if conf.topicsDeprecated != "":
```
Comment (vpavlin) [inline apps/wakunode2/app.nim:552]: How about logging a warning if `conf.topicsDeprecated != ""` ?

**Review by vpavlin — APPROVED** [review]

File: apps/wakunode2/config.nim [hunk 2]
```diff
@@ -207,11 +205,16 @@ type
       defaultValue: false
       name: "keep-alive" }: bool
 
-    topics* {.
-      desc: "Default topics to subscribe to (space separated list)."
-      defaultValue: "/waku/2/default-waku/proto"
+    topicsDeprecated* {.
+      desc: "Default topics to subscribe to (space separated list). DEPRECATED: please use repeated --topic argument instead."
+      defaultValue: ""
```
Comment (alrevuelta) [inline apps/wakunode2/config.nim:210]: should we keep the `/waku/2/default-waku/proto` here? otherwise old deployments using "topics" will break?

**Review by jm-clius — APPROVED** [review]
Thanks! After this is merged the next step is to change this across infra repos, make sure we include this in the release notes as an upgrade instruction and then remove the deprecated option.

**Review by alrevuelta — CHANGES_REQUESTED** [review]
thanks for this, left a comment which i think its important, otherwise we will be introducing breaking changes.

and out of curiosity, why from file its like this? i find it confusing that the behaviour is different. cant we have the same? topic=a, topic=b?
`topic = [ "t1" "t2" "t3" ]`

mind having a look on how `staticnode(s)` is implemented?

File: apps/wakunode2/app.nim (hunk 1 above)
Comment (Ivansete-status) [reply apps/wakunode2/app.nim:552]: Good point! I'll add it

Comment (alrevuelta) [reply apps/wakunode2/app.nim:552]: agree.

```
if conf.topicsDeprecated != "" and conf.topics.len !=0.
  error

if conf.topicsDeprecated != "":
  warn
```

File: apps/wakunode2/config.nim (hunk 2 above)
Comment (jm-clius) [reply apps/wakunode2/config.nim:210]: Oh! Important point that I missed. @alrevuelta is right here. I think we should keep the default in both config options.

Comment (Ivansete-status) [reply apps/wakunode2/config.nim:210]: It shouldn't break because if nothing is specified, the default pubsub topic will get used.

**Comment by Ivansete-status** [comment]
> thanks for this, left a comment which i think its important, otherwise we will be introducing breaking changes.
> 
> and out of curiosity, why from file its like this? i find it confusing that the behaviour is different. cant we have the same? topic=a, topic=b? `topic = [ "t1" "t2" "t3" ]`
> 
> mind having a look on how `staticnode(s)` is implemented?

This behavior comes from 'confutils'. It behaves equally for any other repeatable parameter, i.e. `staticnodes`.

File: apps/wakunode2/config.nim (hunk 2 above)
Comment (alrevuelta) [reply apps/wakunode2/config.nim]: you are right:
https://github.com/waku-org/nwaku/blob/493f082975e7b1bef231acf0299453d645388b8e/apps/wakunode2/app.nim#L559

but perhaps a bit convoluted? can see you applied the changes thanks!

**Review by alrevuelta — APPROVED** [review]
now lgtm thanks!

File: apps/wakunode2/app.nim (hunk 1 above)
Comment (Ivansete-status) [reply apps/wakunode2/app.nim]: Now, if `topicsDeprecated` is given, it prints a `warn`. If both `topicsDeprecated` and `topics` are given, then an error is returned and the execution is halted.

**Pull request merged**